features/signal/external/feature.java

```diff
@@ -0,0 +1,53 @@
+package signal.external;
+
+import io.temporal.activity.ActivityInterface;
+import io.temporal.sdkfeatures.Feature;
+import io.temporal.sdkfeatures.Run;
+import io.temporal.sdkfeatures.Runner;
+import io.temporal.workflow.SignalMethod;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowInterface;
+import io.temporal.workflow.WorkflowMethod;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+@WorkflowInterface
+@ActivityInterface
+public interface feature extends Feature {
+  @SignalMethod
+  void externalSignal(String result);
+
+  @WorkflowMethod
+  public String workflow();
+
+  class Impl implements feature {
+    private static final String SIGNAL_DATA = "Signaled!";
+    private String workflowResult;
+
+    @Override
+    public String workflow() {
+      Workflow.await(() -> workflowResult != null);
+      return workflowResult;
+    }
+
+    @Override
+    public void externalSignal(String result) {
+      workflowResult = result;
+    }
+
+    @Override
+    public Run execute(Runner runner) throws Exception {
+      final var run = runner.executeSingleParameterlessWorkflow();
```

Comment:
Nit: probably don't need `final`

Reply:
I'm a final zealot. But my kind is dying.